irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -185,20 +185,14 @@ namespace torii {
     std::string client_id =
         (client_id_format % context->peer() % hash.toString()).str();
 
-    // in each round, increment the round counter, showing number of consecutive
-    // rounds without status update; if it becomes greater than some predefined
-    // value, stop the status streaming
-    std::atomic_int round_counter{0};
-    consensus_gate_->onOutcome().subscribe(
-        [this, &subscription, &round_counter](const auto &) {
-          auto new_val = round_counter.load() + 1;
-          if (new_val >= maximum_rounds_without_update_) {
-            subscription.unsubscribe();
-          } else {
-            round_counter++;
-          }
-        });
+    auto consensus_gate_observable =
+        consensus_gate_objects_
+            // a dummy start_with lets us don't wait for the consensus event
+            // on further combine_latest
+            .start_with(ConsensusGateEvent{});
 
+    boost::optional<iroha::protocol::TxStatus> last_tx_status;
```

Comment:
I am strongly not sure about thread safety of this variable - it is captured by reference in two lambdas below, which can cause race conditions. Either thread-safe variant of optional should be used (if exists) or operations must be under mutex.
Also there's another way - atomic shared_ptr

Reply:
Can be just std::atomic, since TxStatus is just a enum. But maybe it's possible just merge `take_while` and `filter` combinators?